### PR TITLE
Added error annotation for both format and run buttons

### PR DESCRIPTION
**Pull request by Quarkz14**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -117,20 +117,20 @@
...
+	"os"
 )
 
 templ greeting(name string, imgURL string) {
-  <div style="background-color: lightblue; padding: 20px;">
-    <h1>Hello, { name }</h1>
-    <img src={ imgURL } alt="Welcome image" style="width: 200px; height: auto;" />
-    <p>Welcome to our site. We're glad to have you here!</p>
-  </div>
+	<div style="background-color: lightblue; padding: 20px;">
```
Comment (LotusEngineer) [inline public/index.html:124]: Doesn't indenting like this cause errors if you load the page and just try to run? it had caused it to misbehave for me in the past (I'll give it a little test on my local I was just curious if you'd noticed the same behaviour and maybe your IDE/formatter had just done this)

Comment (Quarkz14) [reply public/index.html:124]: Not for me, @joerdav and I noticed before that the example when we implemented the format button got formatted. So now just went to the example and pasted the formatted code and it seems to respect it. Curious though if it does break for you

Comment (LotusEngineer) [reply public/index.html:124]: It was github's diff being weird, once i pull it down there's no indentation at all on that example code, can ignore.

File: public/script.js [hunk 2]
```diff
@@ -24,8 +24,18 @@ async function formatTempl() {
   formatButton.classList.toggle("loading");
   let templCode = editor.getValue().trim();
 
-  templCode = window.FormatTempl(templCode);
-  editor.setValue(templCode);
+  let templFormattedCode = window.FormatTempl(templCode);
+  if (templFormattedCode.error != undefined) {
```
Comment (LotusEngineer) [inline public/script.js]: I think this could just be 
`if (templFormattedCode.error
) {` ?

File: public/script.js [hunk 3]
```diff
@@ -34,16 +44,27 @@ async function convertTemplToGo() {
   let templCode = editor.getValue().trim();
 
   let goCode = window.ConvertTemplToGo(templCode);
-
+  
+  if (goCode.error != undefined) {
```
Comment (LotusEngineer) [inline public/script.js]: Same as my comment above, think you can use some shorthand here

File: public/script.js [hunk 4]
```diff
@@ -34,16 +44,27 @@ async function convertTemplToGo() {
...
+  editor.setValue(templCode)
   fetch("https://play.golang.org/compile", {
     method: "POST",
-    body: JSON.stringify({ version: 2, body: goCode }),
+    body: JSON.stringify({ version: 2, body: goCode.result }),
   })
     .then((response) => response.json())
     .then((data) => {
-      // console.log("🚀 ~ .then ~ data:", data)
+      console.log("🚀 ~ .then ~ data:", data)
```
Comment (LotusEngineer) [inline public/script.js]: Assume this and the other couple of console logs in here were just for debugging, is it worth removing them at this point?

**Pull request merged**